specs/ScreenCaptureRequested

```diff
@@ -0,0 +1,394 @@
+                                    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo;
+                                    CHECK_FAILURE(args->get_FrameInfo(&frameInfo));
+
+                                    // Frame Name
+                                    wil::unique_cotaskmem_string frameName;
+                                    CHECK_FAILURE(frameInfo->get_Name(&frameName));
+
+                                    // Frame Source
+                                    wil::unique_cotaskmem_string frameSource;
+                                    CHECK_FAILURE(frameInfo->get_Source(&frameSource));
+
+                                    // Frame ID
+                                    wil::com_ptr<ICoreWebView2StagingFrameInfo> frameInfoStaging;
+                                    CHECK_FAILURE(frameInfo->QueryInterface(IID_PPV_ARGS(&frameInfoStaging)));
+                                    UINT32 frameId = 0;
+                                    frameInfoStaging->get_FrameId(&frameId);
+
+                                    // If the host app wants to cancel the request
+                                    CHECK_FAILURE(args->put_Cancel(TRUE));
+
+                                    // Let CoreWebView2 handler know the event is already handled
+                                    PutHandled(args);
+                                    return S_OK;
+                            })
+                            .Get(),
+                        &m_screenCaptureRequestedToken));
+                }
+                return S_OK;
+        }).Get(),
+        &m_FrameCreatedToken));
+}
+
+static void PutHandled(ICoreWebView2ScreenCaptureRequestedEventArgs* args)
+{
+    // In the case of an iframe requesting screen capture permission, the default behavior is
+    // to first raise the ScreenCaptureRequested event off of the CoreWebView2Frame
+    // and invoke it's handlers, and then raise the event off the CoreWebView2
+    // and invoke it's handlers. However, If we set Handled to true on the
+    // CoreWebView2Frame event handler, then we will not raise the
+    // ScreenCaptureRequested event off the CoreWebView2.
+    if (args)
+    {
+        CHECK_FAILURE(args->put_Handled(true));
+    }
+}
+```
+
+## C#: Registering Screen Capture Requested Handler
+```c#
+private WebView2 webView;
+webView.CoreWebView2.ScreenCaptureRequested += (sender, screenCaptureArgs) =>
+{   
+    // Get Frame Info
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo;
+    frameInfo = screenCaptureArgs.frameInfo
+
+    // Frame Name
+    string frameName;
+    frameName = frameInfo.Name;
+
+    // Frame Source
+    string frameSource;
+    frameSource = frameInfo.Source;
+
+    // Frame ID
+    UInt32 frameId = 0;
+    frameId = frameInfo.FrameId;
+
+    // If the host app wants to cancel the request from a specific frame
+    if ( frameId == 1 || frameSource == "https://developer.microsoft.com/en-us/microsoft-edge/webview2/"
```

Comment:
Is frameId of 1 special? Or are you hardcoding as an example for this code sample? If so, a better and more realistic example would be just the URI check and not the frame name or frame ID and a good enough sample for the other cases of checking frame ID or frame name.

Reply:
@david-risney  Yes, it was just hard coded. That makes sense to just use the URI to cancel. What do you mean by "a good enough sample for the other cases of checking frame ID or frame name?"

Reply:
I mean there may be cases where someone wants to check frame ID. If our sample only shows checking the frame URI it should be easy enough for the reader to figure out how to change that to instead check the frame ID. And if we check only the frame URI it should be easier to make a more realistic sample.